server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -52,19 +62,25 @@ private static final class VersionLookup {
         // the tombstone
         private final AtomicLong minDeleteTimestamp = new AtomicLong(Long.MAX_VALUE);
 
+        public static VersionLookup merge(VersionLookup v1, VersionLookup v2) {
```

Comment:
Can we make it explicit that this modifies the `v1` map? I think making it an instance method instead is the easiest way (plus javadoc).

Reply:
I'll improve this. I went through two different ways of accumulating these old maps, and this should be cleanedup.